### PR TITLE
toc nesting audit

### LINKED ISSUE #1895: Audit sidebar nesting
On the left sidebar there are some pages that users might want to get to quickly (like intro to primitives and intro to transpilation) but they are buried in many layers of nesting. I think it would be good to go through our sidebar and make some adjustments overall to reduce nesting where we can. We should keep the top level of nesting but i think the amount of second and third level of nesting could be reduced

### PR DESCRIPTION
**Pull request by abbycross**
Closes #1895

### CODE REVIEW

File: docs/guides/_toc.json [hunk 1]
```diff
@@ -211,67 +201,52 @@
...
-                {
-                  "title": "Represent quantum computers",
-                  "url": "/guides/represent-quantum-computers"
-                }
-              ]
+              "title": "Transpiler stages",
+              "url": "/guides/transpiler-stages"
             },
             {
-              "title": "Advanced transpilation resources",
```
Comment (Eric-Arellano) [inline docs/guides/_toc.json:252]: I think this is still a useful subgroup. I'm not sure if there is a better name like "Custom transpilation and plugins"

File: docs/guides/_toc.json [hunk 2]
```diff
@@ -125,83 +125,73 @@
...
+                "url": "/guides/specify-observables-pauli"
+              },
+              {
+                "title": "The Operator class",
+                "url": "/guides/operator-class"
+              }
+            ]
+          },
+          {
+            "title": "Other circuit-building tools",
```
Comment (javabster) [inline docs/guides/_toc.json:167]: the "other circuit building tools section" should still be inside the circuits and operators top level section

File: docs/guides/_toc.json (hunk 1 above)
Comment (javabster) [reply docs/guides/_toc.json:252]: yeah im happy to keep this subfolder, I would just move it so it is the very last item (i.e. after the transpiler servcie)

File: docs/guides/_toc.json (hunk 2 above)
Comment (abbycross) [reply docs/guides/_toc.json:167]: oops! Yes indeed, will fix

File: docs/guides/_toc.json (hunk 1 above)
Comment (abbycross) [reply docs/guides/_toc.json]: I'm restoring the subfolder - do we want to rename or keep the current name? @javabster

Comment (javabster) [reply docs/guides/_toc.json]: i think keep the current name, "custom" could also imply custom passmanager stuff which isnt in this advanced section

Comment (abbycross) [reply docs/guides/_toc.json]: confirmed - keeping the current name. Subfolder is restored.

**Review by Eric-Arellano — APPROVED** [review]
This lgtm, although please wait for Abby M. 

My only suggested tweak is to restore Transpiler how it was before outside of flattening "Get started with the Qiskit transpiler" -- I thought it made sense for "Configure preset pass managers" to be a folder name. I'm also not sure why Abby M wants Qiskit Transpiler Service to come before "Advanced transpilation resources"?

Thanks for working on this! It's a great improvement.

**Comment by javabster** [comment]
ok yeah Eric raises a good point, now that we've flattened it its not clear which pages are for custom passmanagers vs using generate preset passmanager. 

Ok this is what i think we should do here:

- Introduction to transpilation
- Transpiler stages
- Transpile with pass managers
- Create a passmanager for dynamical decoupling (move page out of advanced section)
- Configure preset pass managers (dropdown)
- Qiskit transpiler service (dropdown)
- Advanced transpilation resources (dropdown)

with this version the intro pages and the custom passmanager content (i.e. the content we most want to encourage people to read) isnt hidden in a dropdown and are logically grouped together. and its clear which pages are about the preset passmanagers bc theyre in their dedicated section. I want to put the transpiler service above the advanced dropdown because it feels logical to have the most advanced stuff be at the end of the section, and transpiler service is actually for people that are less advanced not more advanced

**Review by beckykd — COMMENTED** [review]
LGTM, but I'll let @javabster approve!

**Review by javabster — APPROVED** [review]
LGTM 🚀 it's so nice to have fewer clicks to find what I need!

**Comment by Eric-Arellano** [comment]
Agreed! Excellent job @abbycross!!!

**Pull request merged**